### PR TITLE
titus - add labels to server group summary so they can be filtered in UI

### PR DESCRIPTION
**Pull request by tomaslin**

### CODE REVIEW

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/ServerGroup.java [hunk 1]
```diff
@@ -14,217 +14,277 @@
...
-  Map<String, Object> getLaunchConfig()
+  public Map<String, Object> getLaunchConfig();
+
+  /**
+   * A collection of attributes describing the tags of this server group
+   *
+   * @return a map containing various tags
+   */
+  @Empty
+  default public Map<String, Object> getTags(){
```
Comment (tomaslin) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/ServerGroup.java]: converted to java interface so we can add a default method to getTags

Comment (ajordens) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/ServerGroup.java]: needs space between `()` and `{` ... your editor settings must be off

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/ServerGroup.java [hunk 2]
```diff
@@ -14,217 +14,277 @@
...
  */
-interface ServerGroup {
+public interface ServerGroup {
   /**
    * The name of the server group
    *
    * @return name
    */
-  String getName()
+  public String getName();
```
Comment (ajordens) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/ServerGroup.java]: I'd remove the `public` modifiers, not needed for an interface.

File: clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/model/TitusServerGroup.groovy [hunk 3]
```diff
@@ -158,6 +158,11 @@ class TitusServerGroup implements ServerGroup, Serializable {
   }
 
   @Override
+  Map<String, Object> getTags(){
```
Comment (ajordens) [inline clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/model/TitusServerGroup.groovy]: `(){` formatting

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy [hunk 4]
```diff
@@ -153,8 +154,14 @@ class ServerGroupController {
       securityGroups = serverGroup.getSecurityGroups()
       loadBalancers = serverGroup.getLoadBalancers()
       if (serverGroup.launchConfig) {
-        instanceType = serverGroup.launchConfig.instanceType
+        if(serverGroup.launchConfig.instanceType){
+          instanceType = serverGroup.launchConfig.instanceType
+        }
       }
+      if(serverGroup.tags){
```
Comment (ajordens) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy]: formatting

**Pull request merged**